### PR TITLE
Error handling

### LINKED ISSUE #65: Catch pallet initialization errors
### Steps to reproduce
1. Run a pallet that raises an exception in `__init__`

### Expected behavior
I would expect forklift to complete successfully and show the error raised for that pallet.

### Actual behavior
Forklift chokes and crashes without processing any subsequent pallets.

I think that we should probably wrap [this code](https://github.com/agrc/forklift/blob/eaf951e0067c3276bd7d512d7a7cdd4a989d460a/src/forklift/cli.py#L94-L97) in a try/except. On except we may need to create a fake pallet with the appropriate `success` tuple.

### PR DESCRIPTION
**Pull request by steveoh**
The global handler will catch errors in `__init__` but it will not find bad `imports`. I've updated the logging to spit out warnings so you at least have an error message to go by. 

closes #65

### CODE REVIEW

**Comment by stdavis** [comment]
I think that this is great. But it doesn't solve #65. Forklift still short circuts if an unhandled exception is raised in `Pallet:__init__`. I think that we need a try/except around [these lines](https://github.com/agrc/forklift/blob/eaf951e0067c3276bd7d512d7a7cdd4a989d460a/src/forklift/cli.py#L94-L97).

**Comment by steveoh** [comment]
You'll need to add configration to the config to run this.

File: readme.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ Interacting with forklift is done via the [command line interface](src/forklift/
 - `notify` A list of emails that will be sent the summary report each time `forklift lift` is run (assuming `sendEmails: true`).
 - `repositories` A list of repositories (`<owner>/<name>`) that will be cloned/updated into the warehouse folder.
 - `copyDestinations` A list of folders that you want any data defined in `Pallet.copy_data` to be copied to upon successful processing of the pallet.
+- `configuration` A string of configurations (`Production`, `Staging`, Dev`) that are passed to build to allow a pallet to use different settings based on how forklift is being run. Defaults to `Production`
```
Comment (stdavis) [inline readme.md]: Missing a "`" on this line...

File: src/forklift/__main__.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 Usage:
     forklift config init
+    forklift config open
```
Comment (stdavis) [inline src/forklift/__main__.py]: This should be added to "Examples" below.

Comment (stdavis) [inline src/forklift/__main__.py]: Addressed with: 266e81c

File: readme.md (hunk 1 above)
Comment (stdavis) [inline readme.md]: Addressed with: 0bba2e7

**Comment by stdavis** [comment]
This branch has a bunch of failing tests for me...

**Comment by steveoh** [comment]
Are they config related?I haven't run the tests yet.

**Comment by stdavis** [comment]
Looks like a few are related to the config. Others are related to the `and pallet.success[0]` check. I'm fixing them...

**Comment by stdavis** [comment]
This LGTM but maybe @steveoh should checkout my commits before it gets merged.

**Pull request merged**